### PR TITLE
Add selector-pseudo-class-blacklist rule

### LINKED ISSUE #1119: selector-pseudo-class-black/whitelist
From [a comment](https://css-tricks.com/stylelint/#comment-1601121) on the css-tricks article.

Funnily enough, I was thinking we should probably have a rule for (dis)allowing specific pseudo-class last night when working through `selector-no-qualifying-type`. And then that comment appeared :)

- [x] Name: `selector-pseudo-class-whitelist`
- [x] Name: `selector-pseudo-class-blacklist`
- Primary options: `["string", "/regex/"]`
- Secondary options: none
- Message: `Unexpected pseudo-class selector "${selector}"`.

Any objections to the proposals?

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/33266902-selector-pseudo-class-black-whitelist?utm_campaign=plugin&utm_content=tracker%2F9282518&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F9282518&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by jeddy3**
Closes #1119

### CODE REVIEW

File: src/rules/selector-pseudo-class-blacklist/__tests__/index.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    code: "a:last-of-child {}",
+    message: messages.rejected("last-of-child"),
+    line: 1,
+    column: 2,
+  } ],
+})
+
+testRule(rule, {
+  ruleName,
+  config: ["variable"],
```
Comment (davidtheclark) [inline src/rules/selector-pseudo-class-blacklist/__tests__/index.js]: This should be `[["variable"]]`, right?

**Comment by davidtheclark** [comment]
LGTM, just the one question then merge away.

File: src/rules/selector-pseudo-class-blacklist/__tests__/index.js (hunk 1 above)
Comment (jeddy3) [inline src/rules/selector-pseudo-class-blacklist/__tests__/index.js]: The white/blacklist rules accept a string for their primary option, so this valid. I can change it to a single item array if you like though?

Comment (davidtheclark) [inline src/rules/selector-pseudo-class-blacklist/__tests__/index.js]: Ok.

**Pull request merged**